codec/src/main/java/io/netty/handler/codec/marshalling/MarshallingDecoder.java

```diff
@@ -81,7 +81,7 @@ protected Object decode(ChannelHandlerContext ctx, ByteBuf in) throws Exception
     }
 
     @Override
-    protected ByteBuf extractFrame(ChannelHandlerContext ctx, ByteBuf buffer, int index, int length) {
+    protected ByteBuf extractFrame(ByteBuf buffer, int index, int length) {
```

Comment:
Looked into. At now a param `ctx` is needless. I think it will be correct to deprecate a current `extractFrame` (with `ctx`) in the `LengthFieldBasedFrameDecoder`, make new method without `ctx`, call a new method from the old, and override only new method in the `MarshallingDecoder`.